### PR TITLE
Implemented local search with lunr.js

### PR DESCRIPTION
**Pull request by ochrons**
JS loading is deferred until after page render.

Note: .json compression must be enabled before merging this due to the large size of the search index.json

### CODE REVIEW

File: _includes/footer.html [hunk 1]
```diff
@@ -15,7 +15,9 @@
                 <div class="hline-w"></div>
                 <a class="grey-text text-lighten-3" href="https://gitter.im/scala-js/scala-js">Gitter chat</a><br/>
                 <a class="grey-text text-lighten-3" href="http://stackoverflow.com/questions/tagged/scala.js">Stackoverflow</a><br/>
-                <a class="grey-text text-lighten-3" href="https://groups.google.com/forum/?fromgroups#!forum/scala-js">Mailing list</a>
+                <a class="grey-text text-lighten-3" href="https://groups.google.com/forum/?fromgroups#!forum/scala-js">Mailing list</a><br/>
+                <a class="grey-text text-lighten-3" href="http://twitter.com/scala_js">Twitter <i aria-hidden="true" class="fa fa-twitter"></i></a><br/>
+                <a href="http://github.com/scala-js/scala-js">GitHub <i aria-hidden="true" class="fa fa-github"></i></a>
```
Comment (sjrd) [inline _includes/footer.html]: This looks like it should be in a separate PR.

File: _includes/js.html [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+function downloadJSAtOnload() {
+    var element = document.createElement("script");
+    element.src = "{{ javascriptTag | replace: '<script type="text/javascript" src="', '' | replace: '"></script>', '' }}";
+    document.body.appendChild(element);
+}
+if (window.addEventListener)
+    window.addEventListener("load", downloadJSAtOnload, false);
+else if (window.attachEvent)
+    window.attachEvent("onload", downloadJSAtOnload);
+else window.onload = downloadJSAtOnload;
```
Comment (sjrd) [inline _includes/js.html]: Indentation should be 2 spaces to be consistent with the rest of the codebase.

Comment (sjrd) [inline _includes/js.html]: What does this change do, exactly? Is it related at all to this PR?

File: _includes/nav.html [hunk 3]
```diff
@@ -16,8 +16,7 @@
         <li {% if page.url contains "/libraries" %}class="active"{% endif %}><a href="{{"/libraries/" | prepend: site.baseurl }}">LIBRARIES</a></li>
         <li {% if page.url contains "/community" %}class="active"{% endif %}><a href="{{"/community/" | prepend: site.baseurl }}">COMMUNITY</a></li>
         <li {% if page.url contains "/news" %}class="active"{% endif %}><a href="{{"/news/" | prepend: site.baseurl }}">NEWS</a></li>
-        <li><a href="http://github.com/scala-js/scala-js"><i aria-label="github" style="font-size: 20px" class="fa fa-github"></i></a></li>
-        <li><a href="http://twitter.com/scala_js"><i aria-label="twitter" style="font-size: 20px" class="fa fa-twitter"></i></a></li>
+        <li><a href="#" id="search-icon"><i style="font-size: 20px" class="fa fa-search"></i></a></li>
```
Comment (sjrd) [inline _includes/nav.html]: Ah, I see you've killed the GitHub and Twitter logos from the nav bar ... That explains why they appear in the footer but ... why remove them from the nav bar?

File: _includes/footer.html (hunk 1 above)
Comment (ochrons) [inline _includes/footer.html]: This is here because Twitter and Github links were removed from the top menu to make space for the search icon.

File: community/gethelp.md [hunk 4]
```diff
@@ -1,21 +0,0 @@
----
-layout: community
-title: Get help!
----
```
Comment (sjrd) [inline community/gethelp.md]: This deletion should definitely be in a separate PR.

File: search_data.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+{% endfor %}
+{% for post in site.posts %}
+  {
+  "href": "{{ post.url }}",
+  "title": "{{ post.title | xml_escape }}",
+  "news": "{{ post.content | replace: '><', '> <' | strip_html | tokenize }}"
+  }
+  {% unless forloop.last %},{% endunless %}
+{% endfor %}
+{% endlunrindex %}
```
Comment (sjrd) [inline search_data.json]: Missing EOL at EOF

**Comment by sjrd** [comment]
When I load the page, the `search_data.json` file is automatically downloaded. Could we make it so that it is only loaded when the user clicks the Search button?
That way, we avoid downloading that giant file for users on mobile that don't want to use Search at all (would often happen if they click on a link to a release news or something).

File: _includes/nav.html (hunk 3 above)
Comment (ochrons) [inline _includes/nav.html]: It's been awhile but I think it was due to mobile version not having enough space for all menu items.

Comment (sjrd) [inline _includes/nav.html]: Hum ... if I add them back, I don't have any problem, here. And still plenty of space to fill.

**Comment by sjrd** [comment]
UI-wise, the interface is confusing. When I type in something, I have to switch to the mouse to go and click on a link. I would like to use up/down arrows to select things + Enter to go. Enter could default to the first item.

Also, the search "button" at the right of the input box appears as a link but does nothing. It should be removed if it doesn't do anything.

File: _includes/nav.html (hunk 3 above)
Comment (ochrons) [inline _includes/nav.html]: Could be that there have been some other changes to the top menu since this PR was originally created, but I'll take a look and if there is no space issue, I'll put them back.

**Comment by sjrd** [comment]
That's all, I think.

**Comment by ochrons** [comment]
The search data download is deferred after the page is shown, so it doesn't hinder usage. It's also cached and gzipped, so the penalty is quite low. If it were downloaded only when the search button is clicked, it would be too late from UX point of view.

After gzip it's <100kB in size.

**Comment by sjrd** [comment]
> If it were downloaded only when the search button is clicked, it would be too late from UX point of view.

There's still the time to download it while the user types in something.

> After gzip it's <100kB in size.

I guess that's acceptable, but it still feels like a big hit for all the mobile users who won't ever use the search box, if they hit it on Data usage.

**Comment by sjrd** [comment]
Maybe I'm too paranoid, here.

**Comment by ochrons** [comment]
The search button is indeed superfluous, but you can navigate with Tab+Enter. I'll make an update for this. Arrow key navigation would be nice, but probably requires a bit more work, so can do that as a separate update later.

On mobile the issue is latency, not bandwidth, so downloading search data JIT would not work that well.

Besides, comparing to the old scala-js.org site, I think the overall download size and request count is still way less :)

**Comment by sjrd** [comment]
OK

File: _includes/js.html (hunk 2 above)
Comment (ochrons) [inline _includes/js.html]: This defers loading of search data until the page has been shown to the user.

Comment (sjrd) [inline _includes/js.html]: This should still use 2 spaces indentation (see first comment).

**Comment by sjrd** [comment]
That's all.

**Comment by sjrd** [comment]
LGTM
